sklearn/tests/test_discriminant_analysis.py

```diff
@@ -522,25 +538,23 @@ def test_qda_store_covariance():
     )
 
 
+@pytest.mark.filterwarnings()
```

Comment:
Could please  change this to be more specific on which kind of warning we want to ignore and add a short comment to explain why it's ok to silence those warnings in the context of this test? It's not clear to me.

Reply:
After thinking about this, I thought it would be better to catch the warnings instead of filtering. As far as I could see there was no other test to test the collinearity warning. Also added comments to explain the reasons for the warnings. Thanks for the comment @ogrisel!